### PR TITLE
Move to PHPUnit 9

### PR DESCRIPTION
**Pull request by jordisala1991**
This PR does 4 things:

- Ensure only php 7.2 get the phpunit 8.5
- Ensure we only add the allowed_failures build for php8.0 if we didn't set php 8 on allowed versions.
- Set default phpunit to 9.4 on all builds
- Remove prophecy install from default.

### CODE REVIEW

File: src/Domain/Value/Branch.php [hunk 1]
```diff
@@ -154,6 +154,11 @@ public function lowestPhpVersion(): PhpVersion
         return reset($this->phpVersions);
     }
 
+    public function highestPhpVersion(): PhpVersion
+    {
+        return end($this->phpVersions);
```
Comment (OskarStark) [inline src/Domain/Value/Branch.php:159]: Could you please add a test in `BranchTest` class? Thanks

Comment (OskarStark) [reply src/Domain/Value/Branch.php:164]: I added the method plus a test in https://github.com/sonata-project/dev-kit/pull/1245

File: src/Domain/Value/Branch.php [hunk 2]
```diff
@@ -159,6 +159,11 @@ public function lowestPhpVersion(): PhpVersion
...
+    }
+
```
Comment (OskarStark) [inline src/Domain/Value/Branch.php:166]:
```suggestion
```

File: symfony.lock [hunk 3]
```diff
@@ -168,6 +168,9 @@
     "symfony/dependency-injection": {
         "version": "v4.4.3"
     },
+    "symfony/deprecation-contracts": {
+        "version": "v2.2.0"
+    },
```
Comment (OskarStark) [inline symfony.lock:173]:
```suggestion
```

File: symfony.lock [hunk 4]
```diff
@@ -423,5 +426,8 @@
     },
     "twig/twig": {
         "version": "v3.0.1"
+    },
+    "webmozart/assert": {
+        "version": "1.9.1"
```
Comment (OskarStark) [inline symfony.lock:431]:
```suggestion
```

**Comment by OskarStark** [comment]
Thank you @jordisala1991 🎉

**Comment by VincentLanglet** [comment]
@jordisala1991 I'm not sure it's working properly: https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1183

**Pull request merged**